File: HashTable/C/E771-jewels-and-stones.c
// time:  O(len(J) + len(S))
// space: O(1)

int numJewelsInStones(char* J, char* S) {
    int ht[52] = {0};
    int res = 0;
    for(int i = 0; J[i] != '\0'; i++)
        if(J[i] >= 'a' && J[i] <= 'z')
            ht[J[i] - 'a'] = 1;
        else ht[26 + J[i] - 'A'] = 1;

    for(int i = 0; S[i] != '\0'; i++)
        if(S[i] >= 'a' && S[i] <= 'z'){
            if(ht[S[i] - 'a'])
                res ++;
        }
        else if(ht[26 + S[i] - 'A'])
            res ++;

    return res;
}

File: HashTable/Python3/E771-jewels-and-stones.py
# time:  O(len(S) * T(in))
# space: O(1)

class Solution:
    def numJewelsInStones(self, J, S):
        """
        :type J: str
        :type S: str
        :rtype: int
        """
        res = 0
        
        for s in S:
            if s in J:
                res += 1
                
        return res